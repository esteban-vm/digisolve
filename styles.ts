import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
