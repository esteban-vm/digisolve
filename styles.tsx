import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
