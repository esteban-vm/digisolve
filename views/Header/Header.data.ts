import type { NavLinkProps } from './NavLink'
import { uuid } from '@/helpers'

const links: NavLinkProps[] = [
  {
    id: uuid(),
    text: 'Home',
    link: '#',
  },
  {
    id: uuid(),
    text: 'Contact',
    link: 'contact',
  },
  {
    id: uuid(),
    text: 'About',
    link: 'about',
  },
]

export default links
