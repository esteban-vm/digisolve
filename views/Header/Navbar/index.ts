export { default, navLinks } from './Navbar'
