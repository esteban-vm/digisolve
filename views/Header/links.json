[
  {
    "text": "Home"
  },
  {
    "text": "Contact"
  },
  {
    "text": "About"
  }
]
