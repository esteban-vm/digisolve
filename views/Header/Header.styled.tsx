import { styled } from '@/styles'

export const Wrapper = styled.header`
  background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.7)), url('/img/header-img.jpg');
  background-size: cover;
  background-position: center;
  height: 100vh;
  position: relative;

  & h1 {
    font-size: 500%;
    color: #fff;
    font-weight: 400;
    margin-bottom: 20px;
  }

  & h2 {
    font-size: 200%;
    color: #fff;
    font-weight: 400;
  }
`

export const Box = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
`

export const Text = styled.div`
  margin-bottom: 100px;
`
