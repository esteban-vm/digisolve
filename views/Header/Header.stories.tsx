import type { Meta } from '@storybook/react'
import Header from './Header'

export default {
  component: Header,
  title: 'Views/Header',
} as Meta<typeof Header>

export const HeaderExample = () => <Header />
