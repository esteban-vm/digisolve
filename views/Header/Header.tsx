import type { Component } from '@/types'
import Image from 'next/image'
import { useId } from 'react'
import { Waypoint } from 'react-waypoint'
import { useRef } from 'react'
import { Button } from '@/components'
import { styled, mediaQuery } from '@/styles'
import links from './Header.data'
import NavLink from './NavLink'

const HeaderComponent: Component = (props) => {
  const nameID = useId()
  const navbarRef = useRef<HTMLElement>(null)

  const stickNavbar = (action: 'add' | 'remove') => {
    return () => {
      navbarRef.current?.classList[action]('animate__backInDown', 'sticky')
    }
  }

  return (
    <Waypoint onEnter={stickNavbar('remove')} onLeave={stickNavbar('add')} topOffset='200px'>
      <header id='home' aria-labelledby={nameID} {...props}>
        <nav ref={navbarRef} className='animate__animated animate__fast'>
          <Image src='/img/digisolve-logo.png' alt='Digisolve logo' width={216} height={216} />
          <ul>
            {links.map((link) => (
              <NavLink key={link.id} {...link} />
            ))}
          </ul>
        </nav>
        <div className='box'>
          <div className='titles'>
            <h1 id={nameID}>Digital agency</h1>
            <h2>The one stop for all your digital solutions</h2>
          </div>
          <Button text='Get a quote today!' link='contact' full />
        </div>
      </header>
    </Waypoint>
  )
}

const Header = styled(HeaderComponent)`
  background-image: linear-gradient(var(--color-overlay1), var(--color-overlay2)), url('/img/header-img.jpg');
  background-size: cover;
  background-position: center;
  height: 100vh;
  position: relative;
  color: var(--color-white);
  text-transform: capitalize;

  h1 {
    font-size: 500%;
    font-weight: 400;
    margin-bottom: 20px;
  }

  h2 {
    font-size: 200%;
    font-weight: 400;
  }

  img {
    height: 8.5rem;
    width: auto;
    float: left;
  }

  ul {
    float: right;
    margin-top: 4.3rem;
  }

  .box {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
  }

  .titles {
    margin-bottom: 100px;

    ${mediaQuery('sm')} {
      margin-bottom: 50px;
    }
  }
`

export default Header
