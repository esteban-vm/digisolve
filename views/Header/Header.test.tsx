import { render, cleanup, screen, create } from '@/tests'
import { mediaQuery } from '@/styles'
import Header from './Header'

describe('ðŸ§ª HEADER:', () => {
  describe('display tests:', () => {
    /** Header title */
    const name = /^digital agency$/i

    beforeEach(() => {
      render(<Header />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const header = screen.getByRole('banner', { name })
      expect(header).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the navbar', () => {
        const navbar = screen.getByRole('navigation')
        expect(navbar).toBeInTheDocument()
        expect(navbar).toBeVisible()
      })

      it('the title', () => {
        const title = screen.getByRole('heading', { name, level: 1 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the subtitle', () => {
        const subtitle = screen.getByRole('heading', { name: /all your digital solutions$/i, level: 2 })
        expect(subtitle).toBeInTheDocument()
        expect(subtitle).toBeVisible()
      })

      it('the button', () => {
        const button = screen.getByRole('button', { name: /^get a quote today\!$/i })
        expect(button).toBeInTheDocument()
        expect(button).toBeVisible()
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Header />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles on small screen devices', () => {
      expect(tree).toHaveStyleRule('margin-bottom', '50px', { target: '.titles', media: mediaQuery('sm') })
    })
  })
})
