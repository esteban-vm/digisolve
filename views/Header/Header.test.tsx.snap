// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª HEADER: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-image: linear-gradient(var(--color-overlay1), var(--color-overlay2)),url('img/header-img.jpg');
  -webkit-background-size: cover;
  background-size: cover;
  -webkit-background-position: center;
  background-position: center;
  height: 100vh;
  position: relative;
  color: var(--color-white);
  text-transform: capitalize;
}

.emotion-0 h1 {
  font-size: 500%;
  font-weight: 400;
  margin-bottom: 20px;
}

.emotion-0 h2 {
  font-size: 200%;
  font-weight: 400;
}

.emotion-0 img {
  height: 8.5rem;
  width: auto;
  float: left;
}

.emotion-0 ul {
  float: right;
  margin-top: 4.3rem;
}

.emotion-0 .box {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  text-align: center;
}

.emotion-0 .titles {
  margin-bottom: 100px;
}

@media only screen and (max-width: 768px) {
  .emotion-0 .titles {
    margin-bottom: 50px;
  }
}

.emotion-1 {
  display: inline-block;
  margin-right: 40px;
}

.emotion-1 a:link,
.emotion-1 a:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: var(--color-white);
  outline: none;
  -webkit-transition: border-bottom 100ms;
  transition: border-bottom 100ms;
}

.emotion-1 a:hover,
.emotion-1 a:active {
  border-bottom: 2px solid var(--color-primary);
}

.emotion-4 {
  cursor: pointer;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-4:hover {
  background-color: var(--color-dark-primary);
}

<header
  aria-labelledby=":r0:"
  className="emotion-0"
  id="home"
>
  <nav
    className="animate__animated animate__fast"
  >
    <img
      alt="Digisolve logo"
      data-nimg="1"
      decoding="async"
      height={216}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg%2Fdigisolve-logo.png&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg%2Fdigisolve-logo.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fdigisolve-logo.png&w=640&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={216}
    />
    <ul>
      <li
        className="emotion-1"
        id="ca898ec0-e825-11ed-b24d-893619994521"
      >
        <a
          href="#home"
          onClick={[Function]}
          style={{}}
        >
          Home
        </a>
      </li>
      <li
        className="emotion-1"
        id="cd8d31d0-e825-11ed-b24d-893619994521"
      >
        <a
          href="#contact"
          onClick={[Function]}
          style={{}}
        >
          Contact
        </a>
      </li>
      <li
        className="emotion-1"
        id="d01cb650-e825-11ed-b24d-893619994521"
      >
        <a
          href="#about"
          onClick={[Function]}
          style={{}}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="box"
  >
    <div
      className="titles"
    >
      <h1
        id=":r0:"
      >
        Digital agency
      </h1>
      <h2>
        The one stop for all your digital solutions
      </h2>
    </div>
    <button
      className="emotion-4"
      onClick={[Function]}
      type="button"
    >
      Get a quote today!
    </button>
  </div>
</header>
`;
